Ch04/drill4.cpp

```diff
@@ -0,0 +1,76 @@
+#include "../std_lib_facilities.h"
+
+double sum(const vector<double>& v) {
+	double sumofnum = 0.0;
+	for (double i:v) {
+		sumofnum += i;
+	};
+	return sumofnum;
+}
+
+bool supportedUnit(string unit) {
+	vector<string> units = {"m", "cm", "in", "ft"};
+
+	for (string i: units) {
```

Comment:
Consider using `binary_search()` function from algorithm.h instead of writing this loop.

    #include <algorithm>